src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -72,7 +72,8 @@ Copyright (c) .NET Foundation. All rights reserved.
       RestoreDisableParallel="$(RestoreDisableParallel)"
       RestoreConfigFile="$(RestoreConfigFile)"
       RestoreNoCache="$(RestoreNoCache)"
-      RestoreIgnoreFailedSources="$(RestoreIgnoreFailedSources)" />
+      RestoreIgnoreFailedSources="$(RestoreIgnoreFailedSources)"
+      RestoreNoRecurse="$(RestoreNoRecurse)" />
```

Comment:
The default is to recurse, RestoreNoRecurse disables this. I could go with RestoreRecurse and default it to true in the target unless it has been set. Having a property default to true unless set to false in msbuild feels a bit strange to me, but I do agree that the name isn't great. The alternative of RestoreTopLevelEntryPointProjectsOnly is also not great :smile: